common/stream.h

```diff
@@ -522,13 +559,10 @@ inline stream_out_t::stream_out_t()
 template<typename TType>
 inline void stream_out_t::push(const TType& value)
 {
-	if constexpr (std::is_trivial_v<TType> &&
-	              !(std::is_pointer_v<TType> ||
-	                std::is_reference_v<TType>))
+	if constexpr (std::is_trivially_copyable_v<TType>)
 	{
-		integer_t size = outBuffer.size();
-		outBuffer.resize(size + sizeof(TType));
-		memcpy(&outBuffer[size], &value, sizeof(TType));
+		auto& data = reinterpret_cast<const uint8_t(&)[sizeof(TType)]>(value);
+		outBuffer.insert(outBuffer.end(), std::begin(data), std::end(data));
```

Comment:
_Feel free to ignore_

I become a little scared when I see a reference to a C-style array in an expression. I would recommend:
```cpp
    using ByteArray = const uint8_t(&)[sizeof(TType)];

    if constexpr (std::is_trivially_copyable_v<TType>)
    {
        auto& data = reinterpret_cast<ByteArray>(value);
        //...
 ```